knowrob_kautham/src/kauthamwrapper.cpp

```diff
@@ -0,0 +1,93 @@
+#include "knowrob_kautham/knowrob_kautham.h"
+
+void ham_product(double ax, double ay, double az, double aw, double bx, double by, double bz, double bw, double &rx, double &ry, double &rz, double &rw)
+{
+    rx = bw*ax + bx*aw - by*az + bz*ay;
+    ry = bw*ay + bx*az + by*aw - bz*ax;
+    rz = bw*az - bx*ay + by*ax + bz*aw;
+    rw = bw*aw - bx*ax - by*ay - bz*az;
+}
+
+PREDICATE(kautham_blocking_objects, 3)
+{
+    PlTail colliders(PL_A3);
+    PlTail objpose_list(PL_A1), grasppose_list(PL_A2);
+    PlTerm value;
+    // object pose in map
+    objpose_list.next(value); double otx = value;
+    objpose_list.next(value); double oty = value;
+    objpose_list.next(value); double otz = value;
+    objpose_list.next(value); double orx = value;
+    objpose_list.next(value); double ory = value;
+    objpose_list.next(value); double orz = value;
+    objpose_list.next(value); double orw = value;
```

Comment:
Function would be more readable if this goes to a utility predicate, and maybe also if some math library with vectors and so on would be used.